openwisp_monitoring/monitoring/api/views.py

```diff
@@ -0,0 +1,148 @@
+import logging
+from collections import OrderedDict
+
+from django.conf import settings
+from django.http import HttpResponse
+from pytz import timezone as tz
+from pytz.exceptions import UnknownTimeZoneError
+from rest_framework.exceptions import NotFound, PermissionDenied
+from rest_framework.response import Response
+from rest_framework.views import APIView
+from swapper import load_model
+
+from ...monitoring.exceptions import InvalidChartConfigException
+from ..configuration import DEFAULT_DASHBOARD_TRAFFIC_CHART
+
+logger = logging.getLogger(__name__)
+Chart = load_model('monitoring', 'Chart')
+Metric = load_model('monitoring', 'Metric')
+Organization = load_model('openwisp_users', 'Organization')
+
+
+class DashboardTimeseriesView(APIView):
+    def _get_organizations(self):
+        query_orgs = self.request.query_params.get('organization_slug', '')
+        if query_orgs:
+            query_orgs = query_orgs.split(',')
+            org_ids = Organization.objects.filter(slug__in=query_orgs).values_list(
+                'id', flat=True
+            )
+            if len(org_ids) != len(query_orgs):
+                raise NotFound
+            if self.request.user.is_superuser or all(
+                [str(id) in self.request.user.organizations_managed for id in org_ids]
+            ):
+                return org_ids
+            else:
+                raise PermissionDenied
+        else:
+            if self.request.user.is_superuser:
+                return []
+            elif self.request.user.organizations_managed:
+                return self.request.user.organizations_managed
+            else:
+                raise PermissionDenied
+
+    def _get_query_clause(self, field, items):
+        if not items:
+            return ''
+        lookup = []
+        for item in items:
+            lookup.append(f"{field} = '{item}'")
+        return 'AND ({lookup})'.format(lookup=' OR '.join(lookup))
```

Comment:
any logic related to the specific timeseries DB must be moved to the specific timeseries class which serve as an abstraction layer